### PR TITLE
bring code up to speed

### PR DESCRIPTION
**Pull request by DougAnderson444**
After [having some issues](https://github.com/random-access-storage/random-access-web/issues/6) with `random-access-web` I tried to use `random-access-idb` as a workaround which also was problematic, I was seeing that same issues as #6 and #7. I tried to use substack's version too, but also had issues like not being able to write after reloading the browser.

This PR is the start of trying to get this code to work flawlessly in the browser.

I brought some of the code differences between this repo and substack's which will hopefully fix some of the issues (notably #6  and possibly #7 , I haven't got that far yet, I need time to check the tests). Notably [this commit](https://github.com/substack/random-access-idb/commit/747fc5f57e1406d4f238bbb3a23cbb5457152bff). So that code is brought into this R-A-S repo in this pull. 

Also the tests weren't working for me, so I edited the code so that they work with `browser-run`.

I made the v 1.3.0 to reflect these changes.

Hopefully there's more to come.

### CODE REVIEW

**Comment by DougAnderson444** [comment]
The `db.close()` was causing problems with Corestore, Nanoresource appears to be `[closing]` while the next store is trying to write to it. I don't really see a need to close indexedDB anyway, and removing the close (added in the substack master branch) makes the code work. I added a TODO to follow up how we can ensure the db is open if it's been closed, but in my fork I need this to work before I have time to get to that. 

For your merge consideration.

File: index.js
Comment (soyuka) [inline index.js]: undefined is fine right?

File: package.json
Comment (soyuka) [inline package.json]: is this really needed?

**Review by soyuka — COMMENTED** [review]
Also, you comitted example/bundle.js which is quite huge was this by mistake?

File: package.json
Comment (DougAnderson444) [reply package.json]: I'm trying to remember why I added this. I re-ran the tests with the engine removed, passes all tests. So I'll remove accordingly.

File: index.js
Comment (DougAnderson444) [reply index.js]: Good catch, yes undefined is ok. This was from earlier efforts to troubleshoot issues

**Review by soyuka — APPROVED** [review]

**Comment by RangerMauve** [comment]
@substack Would you mind adding me as a collaborator on NPM so I can merge and publish this?

**Comment by RangerMauve** [comment]
Or add @soyuka for that matter. :P

**Comment by soyuka** [comment]
I can merge @RangerMauve :) I was waiting for another approval. I doubt I can bump this on npm though.

**Comment by soyuka** [comment]
Thanks @DougAnderson444 !

**Pull request merged**
